### DISCUSSION TITLE
GAE output adjacency matrix

### DISCUSSION THREAD

**Discussion by therealcyberlord** [pyg-team/pytorch_geometric#3769] [Q&A] [answered]
Does anyone know how to reconstruct the prediction adjacency matrix using the graph autoencoder? When I call model.decode(), it returns a tensor with the shape: [num_edges], whereas the adjacency matrix is in the format [2, num_edges].  I am hoping to visualize the output graph to see if the model worked correctly.

Any help is appreciated.

**Comment by rusty1s** [comment, marked as answer]
The decoder has the `forward_all` method, which can be used to obtain the fully adjacency matrix, see https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/nn/models/autoencoder.py#L33-L44.

**Reply by therealcyberlord** [reply]
Thank you for the info. Do you know any way to get the predicted edge index from the model? I just realized that the adjacency matrix is in the shape of [num_nodes, num_nodes] rather than [2, num_edges]. Do you know if there is any way to do that? Sorry for the confusion.

**Reply by rusty1s** [reply]
You can try to sparsify it:
```python
edge_index = (adj > 0.8).nonzero(as_tuple=False).t()
```
